ui/tests/acceptance/secrets/backend/ssh/configuration-test.js

```diff
@@ -61,32 +66,61 @@ module('Acceptance | ssh | configuration', function (hooks) {
-
     // default has generate CA checked so we just submit the form
-    await click(SES.ssh.sshInput('configure-submit'));
+    await click(SES.ssh.save);
     assert.strictEqual(
       currentURL(),
       `/vault/secrets/${sshPath}/configuration/edit`,
       'stays on configuration form page.'
     );
-
-    await waitFor(SES.ssh.sshInput('public-key'));
-    assert.dom(SES.ssh.sshInput('public-key')).exists('renders the public key input on form page');
-    assert.dom(SES.ssh.sshInput('public-key')).hasClass('masked-input', 'public key is masked');
-
+    // There is a delay in the backend for the public key to be generated, wait for it to complete by checking that the public key is displayed
+    await waitFor(GENERAL.inputByAttr('public-key'));
+    assert.dom(GENERAL.inputByAttr('public-key')).hasText('***********', 'public key is masked');
+    assert
+      .dom(SES.ssh.editConfigSection)
+      .exists('renders the edit configuration section of the form and not the create part');
+    // delete Public key
+    await click(SES.ssh.deletePublicKey);
+    assert.dom(GENERAL.confirmMessage).hasText('This will remove the CA certificate information.');
+    await click(GENERAL.confirmButton);
+    assert.strictEqual(
+      currentURL(),
+      `/vault/secrets/${sshPath}/configuration/edit`,
+      'after deleting public key stays on edit page'
+    );
+    assert.dom(GENERAL.maskedInput('privateKey')).hasNoText('Private key is empty and reset');
+    assert.dom(GENERAL.inputByAttr('publicKey')).hasNoText('Public key is empty and reset');
+    assert
+      .dom(GENERAL.inputByAttr('generate-signing-key-checkbox'))
+      .isNotChecked('Generate signing key is unchecked');
     await click(SES.viewBackend);
     await click(SES.configTab);
     assert
-      .dom(`[data-test-value-div="Public key"] [data-test-masked-input]`)
-      .hasText('***********', 'value for Public key is on config details and is masked');
-    assert
-      .dom(GENERAL.infoRowValue('Generate signing key'))
-      .hasText('Yes', 'value for Generate signing key displays default of true/yes.');
+      .dom(GENERAL.emptyStateTitle)
+      .hasText('SSH not configured', 'after deleting public key SSH is no longer configured');
     // cleanup
     await runCmd(`delete sys/mounts/${sshPath}`);
   });
 
+  test('it displays error if generate Signing key is not checked and no public and private keys', async function (assert) {
+    const path = `ssh-configure-${this.uid}`;
+    await enablePage.enable('ssh', path);
+    await click(SES.configTab);
+    await click(SES.configure);
+    assert
+      .dom(GENERAL.inputByAttr('generate-signing-key-checkbox'))
+      .isChecked('generate_signing_key defaults to true');
+    await click(GENERAL.inputByAttr('generate-signing-key-checkbox'));
+    await click(SES.ssh.save);
+    assert.true(this.flashDangerSpy.calledWith('missing public_key'), 'Danger flash message is displayed');
```

Comment:
This shouldn't be a flash message but an actual error on the page. Stay tuned, it becomes one in the ssh refactor.